bin/pre-push

```diff
@@ -1,16 +1,106 @@
 #!/bin/sh
 
+remote="$1"
+url="$2"
+sha40=0000000000000000000000000000000000000000
+
 echo "\nBy contributing to this project, you license the materials you contribute under the GNU General Public License v2 (or later). All materials must have GPLv2 compatible licenses — see .github/CONTRIBUTING.md for details.\n\n"
 
-current_branch=$(git symbolic-ref HEAD | sed -e 's,.*/\(.*\),\1,')
-if [ 'master' = $current_branch ]
-then
-    read -p "You're about to push ¡¡¡[ master ]!!!, is that what you intended? [y|n] " -r < /dev/tty
-    echo
-    if echo $REPLY | grep -E '^[Yy]$' > /dev/null
-    then
-        exit 0 # push will execute
-    fi
-    exit 1 # push will not execute
-fi
+# We want to know whether this script was called from the CLI or from a GUI app.
+# If it is the later, we would only show the ESLint command to get the report,
+# not the report itself, because GUIs tend to left a small space for pre-commit
+# messages and they are not useful. At least this way, the user can see something.
+#
+# - First, get the caller: the process that executes the commit.
+#	The command tree would be: caller > git > pre-commit script
+#       so we need to get the grandparent of this script
+#
+# - Second, if the caller is not one of the available shells (see /etc/shells),
+#	we are going to assume that it is called from a GUI app.
+
+# caller=$(which $(ps -o comm=  $(ps -o ppid= $(ps -o ppid= $$))))
+caller=$(which $(ps -o comm=  $(ps -o ppid= $PPID)))
+
+is_cli=false
+while read -r line;
+do
+	if [ "$line" = "$caller" ]
+	then
+		is_cli=true
+		break
+	fi
+done < /etc/shells
+
+while read local_ref local_sha remote_ref remote_sha
+do
+	# echo "l_ref $local_ref l_sha $local_sha r_ref $remote_ref r_sha $remote_sha"
```

Comment:
No, but now that JavaScript is mainstream, I think shell scripting is well positioned to become The World's Most Misunderstood Programming Language. I realize I have a tendency to overcomment or leave debug comments -just in case- in shell scripts. This is because, as it is not my day-to-day language, I feel that I may miss something when I review the same code 6 months down the line.